[expo-app-loading][expo] Extract AppLoading to separate module (#10929)
expo/expo: packages/expo-app-loading/README.md

# expo-app-loading

> A React component that tells Expo to keep the app loading screen open if it is the first and only component rendered in your app. Unless the `autoHideSplash` prop is set to `false`, the loading screen will disappear and your app will be visible when the component is removed.

Reviewer: ```suggestion
A React component that tells expo-splash-screen to remain visible if it is the first and only component rendered in your app.
```

> This is incredibly useful to let you download and cache fonts, logos, icon images and other assets that you want to be sure the user has on their device for an optimal experience before rendering and they start using the app.

Reviewer: ```suggestion
This can be useful while download and cache fonts, logos, icon images and other assets that you want to be sure the user has on their device for an optimal experience before rendering and they start using the app. You can alternatively use expo-splash-screen APIs directly - expo-app-loading just wraps them.
```
